### DISCUSSION TITLE
Consider dropping or moving support for docker-notation and docker-generate

### DISCUSSION THREAD

**Discussion by dtzar** [notaryproject/notation#251] [General]
Are you using or plan to use docker-generate or docker-notation in the repository?

Should this be converted to a plugin in a different repository?

[This was brought up in a PR ](https://github.com/notaryproject/notation/pull/250#issuecomment-1188011110) and good to consider by the community and group of maintainers.  We would like feedback and a discussion here.

**Comment by sajayantony** [comment]
This was a prototype. I think we should drop it since we don't have anyone to support or completely review the experience.

**Comment by FeynmanZhou** [comment]
If docker-notation is removed, does it mean users will not need to install Docker as a dependency for notation?

**Reply by dtzar** [reply]
No - this is just a prototype of having `docker` itself having `notation` integrated via an extension.

**Reply by FeynmanZhou** [reply]
Make sense. Thanks for your clarification.

**Comment by dtzar** [comment]
Ok - since it seems this might be easily referenceable for potential future prototyping work, any preference on a branch here (which will get stale) or put it in a separate repo?

**Reply by SteveLasker** [reply]
Putting in a separate repo will give it more visibility than I think we're ready to maintain.
Having a locked branch may suffice, or even a label might suffice. 
I defer to @shizhMSFT, who may say, the code for this isn't worth saving, as it will be easy to duplicate in the future, based on that current state of the repo. 
Or, there's enough code to save, just thinking a branch or label is the right level of "mental space" to maintain.

**Reply by shizhMSFT** [reply]
I’d suggest putting `docker-notation` and `docker-generate` in a separate repo. In the end, docker CLI will have similar stuffs integrated and that repo will be archived.

**Reply by sajayantony** [reply]
I don't think this is needed  since we already have [v0.9.0-alpha.1](https://github.com/notaryproject/notation/releases/tag/v0.9.0-alpha.1) which we can always checkout if we really need to see the code.

**Reply by dtzar** [reply]
We can also use this discussion/reference now to go back and get the code from the v0.9.0-alpha.1 release.  Agree Steve - separate repo will likely give it more visibility than we're ready for right now.

**Comment by SteveLasker** [comment]
What I'm seeing from @shizhMSFT, is there is enough code, but we can maintain that code, without any new work, using the  [v0.9.0-alpha.1](https://github.com/notaryproject/notation/releases/tag/v0.9.0-alpha.1) release. Sounds like a great balance.
LGTM

**Comment by SteveLasker** [comment]
Since this is an discussion, not an issue, do we just delete this? Or, convert to an issue until the code is removed?

**Reply by dtzar** [reply]
Added [this issue ](https://github.com/notaryproject/notation/issues/254) to track the code being removed.
